wdl-doc/src/lib.rs

```diff
@@ -67,282 +170,198 @@ impl Document {
+    pub fn preamble(&self) -> Markup {
+        let preamble = parse_preamble_comments(self.version.clone());
+        Markdown(&preamble).render()
     }
-}
-
-impl Display for Document {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let document_name = content::Heading1::builder()
-            .text(self.name().to_owned())
-            .build();
-        let version = text_content::Paragraph::builder()
-            .text(format!("WDL Version: {}", self.version()))
-            .build();
-
-        let mut options = Options::empty();
-        options.insert(Options::ENABLE_TABLES);
-        options.insert(Options::ENABLE_STRIKETHROUGH);
-        let parser = Parser::new_ext(&self.preamble, options);
-        let mut preamble = String::new();
-        pulldown_cmark::html::push_html(&mut preamble, parser);
 
-        write!(f, "{}", document_name)?;
-        write!(f, "{}", version)?;
-        write!(f, "{}", preamble)
-    }
-}
-
-/// Fetch the preamble comments from a document.
-pub fn fetch_preamble_comments(document: AstDocument) -> String {
-    let comments = match document.version_statement() {
-        Some(version) => {
-            let comments = version
-                .keyword()
-                .syntax()
-                .preceding_trivia()
-                .map(|t| match t.kind() {
-                    wdl_ast::SyntaxKind::Comment => match t.to_string().strip_prefix("## ") {
-                        Some(comment) => comment.to_string(),
-                        None => "".to_string(),
-                    },
-                    wdl_ast::SyntaxKind::Whitespace => "".to_string(),
-                    _ => {
-                        panic!("Unexpected token kind: {:?}", t.kind())
+    /// Render the document as HTML.
+    pub fn render(&self) -> Markup {
+        html! {
+            div {
+                h1 { (self.name()) }
+                h3 { "WDL Version: " (self.version()) }
+                div { (self.preamble()) }
+                div class="flex flex-col items-center text-left"  {
+                    h2 { "Table of Contents" }
+                    table class="border" {
+                        thead class="border" { tr {
+                            th class="" { "Page" }
+                            th class="" { "Type" }
+                            th class="" { "Description" }
+                        }}
+                        tbody class="border" {
+                            @for page in &self.local_pages {
+                                tr class="border" {
+                                    td class="border" {
+                                        a href=(page.0.to_str().unwrap()) { (page.1.name()) }
+                                    }
+                                    td class="border" {
+                                        @match page.1.page_type() {
+                                            PageType::Index(_) => { "TODO ERROR" }
```

Comment:
Can this not be a literal `todo!()`?

Reply:
Unfortunately, that doesn't work within the `maud` macro. Trying to write `unreachable!()` or other such panic gets the entire macro triggered as "unreachable code block" by clippy. I saw some folks online using those panics, but they were also silencing the clippy warning.